packages/webassembly-playground/src/ui/Editor.tsx

```diff
@@ -3,35 +3,39 @@ import * as monaco from "monaco-editor";
 
 export type EditorProps = {
   initialContent?: string;
+  language?: string;
 };
 
 export type EditorRef = {
   getEditor(): monaco.editor.IStandaloneCodeEditor | null;
 };
 
-const Editor = forwardRef<EditorRef, EditorProps>(({ initialContent }, ref) => {
-  const editorRef = useRef<monaco.editor.IStandaloneCodeEditor | null>(null);
-  const containerRef = useRef<HTMLDivElement>(null);
-
-  useImperativeHandle(ref, () => ({
-    getEditor: () => editorRef.current,
-  }));
-
-  useEffect(() => {
-    const editor = monaco.editor.create(containerRef.current!, {
-      value: initialContent,
-      automaticLayout: true,
-      tabSize: 2,
-      minimap: { enabled: false },
-    });
-    editorRef.current = editor;
-
-    return () => {
-      editor.dispose();
-    };
-  }, [initialContent]);
-
-  return <div ref={containerRef} />;
-});
+const Editor = forwardRef<EditorRef, EditorProps>(
+  ({ initialContent, language }, ref) => {
+    const editorRef = useRef<monaco.editor.IStandaloneCodeEditor | null>(null);
+    const containerRef = useRef<HTMLDivElement>(null);
+
+    useImperativeHandle(ref, () => ({
+      getEditor: () => editorRef.current,
+    }));
+
+    useEffect(() => {
+      const editor = monaco.editor.create(containerRef.current!, {
+        value: initialContent,
+        language,
+        automaticLayout: true,
+        tabSize: 2,
+        minimap: { enabled: false },
+      });
+      editorRef.current = editor;
+
+      return () => {
+        editor.dispose();
+      };
+    }, [initialContent]);
+
```

Comment:
We should update the language of the editor's current model if the `language` prop changed.

Reply:
We need eslint-plugin-react-hooks to detect this.